### ISSUE TITLE
Some marathon jobs cannot be "destroyed" if they still have any running instances.

### ISSUE THREAD

**Issue by solarkennedy** [mesosphere/marathon#725]
I have a marathon instance, and some jobs running. Each is scaled to a single instance.

When I click the "destroy" button, nothing seems to happen on the gui. Every once in a while I can see a deployments badge flash to 1, but the job remains on the task list.

Here are the marathon logs for that event, some whitespace added:
````
[2014-10-20 21:26:06,281] INFO 10.56.5.45 - admin [20/Oct/2014:21:26:06 +0000] "DELETE /v2/apps/cieye.canary.e08e6740ffdb16ccf30c94741a4ed4e1 HTTP/1.1" 200 92 "-" "python-requests/2.4.3 CPython/2.6.7 Linux/3.13.0-34-generic" (mesosphere.chaos.http.ChaosRequestLog:15)

[2014-10-20 21:26:06,340] INFO Compute DeploymentPlan 

from Group(/,Set(AppDefinition(/paasta--test.hhvm.0d8a64d5f0794a05fda3eee461072b27,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(10486),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:hhvm0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:06.988Z), AppDefinition(/cieye.canary.398a94adbe0217a77fe6dc5caec43d2e,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(19628),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:19.695Z), AppDefinition(/paasta--test2.hhvm.018e91e1cf1b9170bf13826c69a4400b,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(16596),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:hhvm0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:57.415Z), AppDefinition(/paasta--test2.canary.7218f7735d2342e4f35a9d579599b962,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(15204),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:54.248Z), AppDefinition(/cieye.main.df7428cc8f3dc4e443cd328caa2a0eae,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(10460),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:16.510Z), AppDefinition(/paasta--test2.main.299b4215764bfd8a97c463ab91d7be3b,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(13041),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:51.032Z), AppDefinition(/mumble.main.732656e729d6f16bf7cc29222665e725,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(file:///root/.dockercfg),Vector(),Vector(10001),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(Volume(/state,/state,RO), Volume(//srv,//srv,RO)),Some(Docker(private:443/services-mumble:jenkins-d62a0a912c36c91ac77af1b8c6cfabfed819ed07,Some(BRIDGE),Some(Vector(PortMapping(8888,0,0,tcp))))))),Set(),Set(),UpgradeStrategy(1.0),2014-10-20T21:22:47.771Z), AppDefinition(/cieye.canary.a2f6b9644d277df62e645f3d06893677,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(file:///root/.dockercfg),Vector(),Vector(10000),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(Volume(/state,/state,RO), Volume(//srv,//srv,RO)),Some(Docker(private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,Some(BRIDGE),Some(Vector(PortMapping(8888,0,0,tcp))))))),Set(),Set(),UpgradeStrategy(1.0),2014-10-09T22:18:48.933Z)),Set(),Set(),2014-10-09T22:18:48.933Z)

to   Group(/,Set(AppDefinition(/paasta--test.hhvm.0d8a64d5f0794a05fda3eee461072b27,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(10486),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:hhvm0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:06.988Z), AppDefinition(/cieye.canary.398a94adbe0217a77fe6dc5caec43d2e,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(19628),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:19.695Z), AppDefinition(/paasta--test2.hhvm.018e91e1cf1b9170bf13826c69a4400b,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(16596),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:hhvm0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:57.415Z), AppDefinition(/paasta--test2.canary.7218f7735d2342e4f35a9d579599b962,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(15204),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:54.248Z), AppDefinition(/cieye.main.df7428cc8f3dc4e443cd328caa2a0eae,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(10460),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:17:16.510Z), AppDefinition(/paasta--test2.main.299b4215764bfd8a97c463ab91d7be3b,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(),Vector(),Vector(13041),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(),Some(Docker(docker:///private:443/soa_paasta_test:0.1,None,None)))),Set(),Set(),UpgradeStrategy(1.0),2014-09-30T18:16:51.032Z), AppDefinition(/mumble.main.732656e729d6f16bf7cc29222665e725,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(file:///root/.dockercfg),Vector(),Vector(10001),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(Volume(/state,/state,RO), Volume(//srv,//srv,RO)),Some(Docker(private:443/services-mumble:jenkins-d62a0a912c36c91ac77af1b8c6cfabfed819ed07,Some(BRIDGE),Some(Vector(PortMapping(8888,0,0,tcp))))))),Set(),Set(),UpgradeStrategy(1.0),2014-10-20T21:22:47.771Z), AppDefinition(/cieye.canary.a2f6b9644d277df62e645f3d06893677,None,Some(Vector()),None,Map(),1,1.0,100.0,0.0,,Set(),Vector(file:///root/.dockercfg),Vector(),Vector(10000),false,1000 milliseconds,1.15,Some(Container(DOCKER,Vector(Volume(/state,/state,RO), Volume(//srv,//srv,RO)),Some(Docker(private:443/services-cieye:jenkins-231fe2fa956514e2269edee69345b2009010464f,Some(BRIDGE),Some(Vector(PortMapping(8888,0,0,tcp))))))),Set(),Set(),UpgradeStrategy(1.0),2014-10-09T22:18:48.933Z)),Set(),Set(),2014-10-09T22:18:48.933Z) (mesosphere.marathon.upgrade.DeploymentPlan$:87)

[INFO] [10/20/2014 21:26:06.342] [marathon-akka.actor.default-dispatcher-11] [akka://marathon/user/MarathonScheduler/$a] Deployment of / successful

[INFO] [10/20/2014 21:26:06.342] [marathon-akka.actor.default-dispatcher-17] [akka://marathon/user/MarathonScheduler/$a/UpgradeManager] Removing bac5c93d-9d09-4e62-8ede-77b42bb4d5e9 from list of running deployments

````

The From and To group is identical? Then the deployment is removed. The task is *still* in the gui.

However, I *can* suspend the instances to 0, and then delete the job, and it works. 

Any ideas on what is going on? What additional information can I provide to diagnose this?

marathon 7.3 (3 masters)
docker 0.20.1 with native docker stuff
all in one big group ("/")

cc @EvanKrall @mrtyler

**Comment by rqc** [comment]
Hi, does anyone have any news on this issue? This seems like an important scenario to support for automated deployment and removal of apps. I am experiencing the same issue. I have a script that creates and deletes apps with multiple requests to the REST API. Some jobs are left dangling and they cannot be killed. I tried the scaling-down workaround and it worked sometimes. I have a similar script for posting jobs on Chronos and it works perfectly.

**Comment by drexin** [comment]
@rqc Could you try 0.8.0-RC1 and see if this problem is still present?

**Comment by rqc** [comment]
@drexin Thanks for your response. I was able to clone and build 0.8.0-RC1 and was able to reproduce the problem. :disappointed: 

Now, let me provide a bit more information on the driver script that is deleting the apps. It is a python script that uses futures to parallelize the delete requests to marathon (using 4 workers). I thought that maybe marathon was having problems processing the requests if they come in at the same time (multithreading). In fact, 4 delete requests come in, but not all of them get processed. 

I changed my algorithm to run with just 1 worker and placed a 5 second delay between each delete request and it worked! I've tried it a few times and it has worked consistently.

Perhaps there is an issue in how the requests are being being dispatched to Mesos or how Marathon is queuing those requests so that they get processed correctly?

Thanks for your help!

**Comment by drexin** [comment]
@rqc What is the state of Marathon when you call your script? Are there any running deployments? Do you call it with `?force=true`?

**Comment by rqc** [comment]
@drexin No running deployments that I'm aware of, and just 1 task per app running. Not calling with `?force=true`.

**Comment by solarkennedy** [comment]
I can confirm this is still present on 8.1

It still happens regardless of current deployments or not. ?force=true doesn't have any extra effect.

Suspending the app and then deleting it makes it go away immediately.

**Comment by aquamatthias** [comment]
This should be fixed with Marathon 0.13.0. Please reopen, if this is a Marathon problem, that still exists.
